civis_jupyter_notebooks/assets/extensions/terminal.js

```diff
@@ -9,6 +9,7 @@ define(['base/js/namespace'], function(Jupyter) {
         // stop pop up from displaying
         window.onbeforeunload = function(){}
         window.location.href = "/terminals/1";
+        env.notebook.set_dirty(false);
```

Comment:
seems like this should happen before we change the location? I'm not sure it's guaranteed to execute otherwise..